### DISCUSSION TITLE
Fastify Plugin: Ability to modify spec. on the fly

### DISCUSSION THREAD

**Discussion by qqilihq** [scalar/scalar#3433] [Q&A]
We're currently using [fastify-swagger-ui](https://github.com/fastify/fastify-swagger-ui) but I'm enjoying scalar very much - great work 👍 

In our current setup using `fastify-swagger-ui` we have implemented a technique which allows to show an extended spec for “pro users”. For that, they load the spec via `api.example.com/documentation` and modify the URL in the Swagger UI URL bar from `https://api.example.com/documentation/json` to `https://api.example.com/documentation/json?expert=true`.

This approach is implemented using fastify-swagger-ui’s [`transformSpecification`](https://github.com/fastify/fastify-swagger-ui?tab=readme-ov-file#transformspecification):

```typescript
await fastify.register(fastifySwaggerUi, {
  routePrefix: '/documentation',
  transformSpecification: (swaggerObject, request) => {
    const expertSpec = request.query.expert === 'true';
    if (expertSpec) {
      return swaggerObject;
    } else {
      return trimDownSpec(swaggerObject);
    }
  },
  // …
);
```

I'm currently wondering how to best map this to scalar:

1. Is there some possibility included to detect a query parameter when accessing the spec?
2. Is there an equivalent of transformSpecification? There's a spec factory, but it doesn't provide any further context:

```typescript
await fastify.register(scalarApiReference, {
  routePrefix: '/documentation',
  configuration: {
    spec: {
      // called upoon each request, but no request-specific parameters or context available which I could use
      content: () => trimDownSpec(fastify.swagger())
    }
  },
  // …
);
```

3. Any better ideas how to implement this?

**Comment by hanspagel** [comment]
Rad! Let’s add this: #3434
